poc/h01_callback_reenter.py

```
@@ -0,0 +1,74 @@
+
+Impact: staker reward vault (GRC20) double-paid at old tick/acc (pre-swap) if tick crossed.
+Launchpad TransferLeft similarly: shared OBL/rewardToken balance drained via sibling project accounting before pool balance settles.
+
+Why not fixed by #1354: #1354 added assertPoolUnlocked only to pool.UpgradeImpl (governance path). Callback→staker path uncovered.
+
+Fix: gate staker entrypoints CollectReward/UnStakeToken/StakeToken + launchpad TransferLeft with pool lock check,
+or add reentry depth guard in safeSwapCallback (forbid vm.Call to staker/launchpad while lock held).
+"""
+
+print(evidence)
+with open("/tmp/h01_evil_realm.gno","w") as f:
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/h01_evil_realm.gno"

(S108)